Check PassiveIP eq External IP, Eximstats to M-M-M-Monster Logs, /var/cpanel/tomcat.options -Xss2m

cPPietila opened the pull request:
3 commits to alter the functionality of SSP.

- Ignore -Xss2m in /var/cpanel/tomcat.options
- Add eximstats_db.sqlite3 to M-M-M-MONSTER LOG
- Add checking that FTP announces correct IP

cPanelChrisD commented on ssp:
```diff
@@ -2505,8 +2514,28 @@ sub print_ftpserver_info {
+
+    if ($passiveip) {
+        if ( $passiveip ne $EXTERNAL_IP_ADDRESS ) {
+            if ( defined( $CPCONF{'ftpserver'} ) ) {
```
These if blocks should be combined:
if ( defined( $CPCONF{'ftpserver'} ) and ( $passiveip ne $EXTERNAL_IP_ADDRESS ) ) {

davelcpanelnet-zz replied:
You probably want to enforce the logic here even further as well:
```
if ( exists $passiveip && $passiveip ne $EXTERNAL_IP_ADDRESS && defined( $CPCONF{'ftpserver'} )
```

cPanelChrisD commented on ssp:
```diff
@@ -2505,8 +2514,28 @@ sub print_ftpserver_info {
+        if ( $passiveip ne $EXTERNAL_IP_ADDRESS ) {
+            if ( defined( $CPCONF{'ftpserver'} ) ) {
+                if ( $CPCONF{'ftpserver'} eq 'proftpd' ) {
+                    $passiveiptext = "MasqueradeAddress ( " . $passiveip . " ) doesn't match license IP";
```
Concatenate this text on the end of $passivetext, it's not necessary to put it in a separate $passiveiptext.  6 lines of code can be dropped by doing so, reducing complexity.

cPanelChrisD reviewed (changes requested):
Please create and assign JIRA TECH cases to yourself for each issue that you will be addressing in SSP.  For example, you can assign TECH-296 to yourself and mark it as completed (with a link to the matching commit in your personal fork of the repo, please).

This is also going to conflict with the pull request I sent in yesterday.  That merge will additionally begin the requirement that SSP passes perlcritic tests defined in the Makefile with the provided perlcritic profile, in addition to being tidy.  After that pull is merged, you can "make test" to just run the tests, and "make tidy" will run both test and tidy targets.

davelcpanelnet-zz commented:
Please make the changes requested and be sure to note the TECH-* for proper visibility.

This pull request will be left open so please feel free to make those updates and even make a few changes should you wish.

cPPietila committed:
TECH-297 Add check that FTP announces correct IP

Adds text to the FTP Server line if the MasqueradeAddress (ProFTPd) or ForcePassiveIP (Pure-FTPd) does not match the external IP address detected on port 80.

cPPietila committed:
TECH-298 Add eximstats sqlite to M-M-M-MONSTER LOG

Size parameter set to 5 billion bytes, to respect the previously-monitored-for value.

cPPietila committed:
TECH-296 Ignore -Xss2m in tomcat.options

cPanel added that as a response to a kernel vulnerability fix that broke tomcat. If it's there, and that's all that's there, it's safe to ignore.

cPPietila committed:
Bump version for pull request.

cPPietila commented:
I've gone ahead and made the requested changes.

cPanelChrisD reviewed (approved):
All looks good.  Ready for Dave's review.

davelcpanelnet-zz commented:
Good to go.
